money-input.html

```diff
@@ -185,10 +195,53 @@
         this._formattedValue = "";
       },
 
+      _valuePreProcessing: function() {
+        var signal = ("" + this.value).indexOf("\-") >= 0 ? "-" : "";
+
+        var multiplier = Math.pow(10, this.precision);
+        var value = (this.value * multiplier) / multiplier;
+        var fraction = value % 1;
+        var decFraction = (fraction * multiplier) / multiplier;
+
+        var integer = undefined;
+        var completeNumber = undefined;
+
+        if (decFraction == 0 && this.precision != 0) {
+          completeNumber = signal + value + ".00";
+        }
+        else {
+          integer = Number(this.value) - Number(decFraction);
+          completeNumber = (integer + decFraction);
+
+          completeNumber = completeNumber + '';
+          var respFraction = completeNumber.split('.');
+
+          if (respFraction[1]) {
+            var respFractionLen = respFraction[1].length;
```

Comment:
Maybe, if you could explain the main idea here, I could implement it (so, you would have your issues solved) but in a different way...